beacon-chain/blockchain/chain_info.go

```diff
@@ -53,12 +53,12 @@ type HeadFetcher interface {
 	HeadETH1Data() *ethpb.Eth1Data
 	HeadPublicKeyToValidatorIndex(pubKey [fieldparams.BLSPubkeyLength]byte) (types.ValidatorIndex, bool)
 	HeadValidatorIndexToPublicKey(ctx context.Context, index types.ValidatorIndex) ([fieldparams.BLSPubkeyLength]byte, error)
-	ProtoArrayStore() *protoarray.Store
 	ChainHeads() ([][32]byte, []types.Slot)
 	IsOptimistic(ctx context.Context) (bool, error)
-	IsOptimisticForRoot(ctx context.Context, root [32]byte, slot types.Slot) (bool, error)
+	IsOptimisticForRoot(ctx context.Context, root [32]byte) (bool, error)
 	HeadSyncCommitteeFetcher
 	HeadDomainFetcher
+	ForkChoicer() forkchoice.ForkChoicer
```

Comment:
From my first glance, just returning `ForkChoicer()` seems too broad. I wonder if we need to encapsulate this behavior better. Something I will look into in the following up PRs

Reply:
I think the previous behavior was worse: we exposed the store or even nodes outside of forkchoice, this made some functions outside of forkchoice to depend on the internals of the package. I removed this by not exposing ever the Store() nor Node() and any function (like the debug rpc endpoint) that requires this info, to request it from the forkchoice. At the same time this makes for better encapsulation and also it allows me to have completely different stores as in this PR with the same interface.